Xamarin.Essentials/MediaPicker/MediaPicker.ios.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Foundation;
+using MobileCoreServices;
+using UIKit;
+
+namespace Xamarin.Essentials
+{
+    public static partial class MediaPicker
+    {
+        static async Task<MediaFile> PlatformShowPhotoPickerAsync(MediaPickerOptions options)
+        {
+            if (!UIImagePickerController.IsSourceTypeAvailable(UIImagePickerControllerSourceType.PhotoLibrary))
+                throw new FeatureNotSupportedException();
+            if (!UIImagePickerController.AvailableMediaTypes(UIImagePickerControllerSourceType.PhotoLibrary).Contains(UTType.Image))
+                throw new FeatureNotSupportedException();
+
+            // permission is not required on iOS 11 for the picker
+            if (!Platform.HasOSVersion(11, 0))
+                await Permissions.RequestAsync(PermissionType.Photos);
+
+            var vc = Platform.GetCurrentViewController();
+
+            var picker = new UIImagePickerController();
+            picker.SourceType = UIImagePickerControllerSourceType.PhotoLibrary;
+            picker.MediaTypes = new string[] { UTType.Image };
```

Comment:
Perhaps make this a static readonly to avoid constant array allocation, as done in uwp